### PR TITLE
Jumpbug

### PR DESCRIPTION
**Pull request by twangsriprasoet**
Pasted from UC.
AntarioSDK to OsirisSDK.

### CODE REVIEW

**Comment by twangsriprasoet** [comment]
I accidentally hit the revert. sorry. 😅

**Comment by RyDeem** [comment]
Check this out, i unobfuscated some bools and fixes and stuff, nothing functional, removed some fag's? lmfao

https://gist.github.com/RyDeem/fdcff7f5a4089cbd825c5cec166f7b0e or https://github.com/RyDeem/Osiris/tree/oofers2463

**Comment by notgoodusename** [comment]
replace plocalplayer with localplayer, also why are u forcing bunnyhop to true?

**Comment by newcommerdontblame** [comment]
Can u paste autostrafe optimizer pls

**Comment by twangsriprasoet** [comment]
> replace plocalplayer with localplayer, also why are u forcing bunnyhop to true?
you were right. but i didn't see Bunnyhop forced to turn on while Jumpbug enabled. or i just being stupid. 😓

**Comment by twangsriprasoet** [comment]
> Check this out, i unobfuscated some bools and fixes and stuff, nothing functional, removed some fag's? lmfao
> 
> [RyDeem@5a57eb4](https://github.com/RyDeem/Osiris/commit/5a57eb4b3ddf524562a295572f1c47b7cc732c24)
> 
> https://gist.github.com/RyDeem/fdcff7f5a4089cbd825c5cec166f7b0e or
> https://github.com/RyDeem/Osiris/tree/oofers2463

thanks for the help!

**Comment by twangsriprasoet** [comment]
ahhh i forgot to remove pLocalPlayer..

**Comment by daisuki333** [comment]
do u know a way to create a velocity text thing?

**Comment by twangsriprasoet** [comment]
> do u know a way to create a velocity text thing?

localPlayer->velocity().length2D(); ?

**Comment by daisuki333** [comment]
> > do u know a way to create a velocity text thing?
> 
> localPlayer->velocity().length2D(); ?

how do i make it display it tho? like this 
image

**Comment by NekoRem** [comment]
> > > do u know a way to create a velocity text thing?
> > 
> > 
> > localPlayer->velocity().length2D(); ?
> 
> how do i make it display it tho? like this
> image

You just render text on the bottom of the screen and thats it

**Comment by daisuki333** [comment]
> > > > do u know a way to create a velocity text thing?
> > > 
> > > 
> > > localPlayer->velocity().length2D(); ?
> > 
> > 
> > how do i make it display it tho? like this
> > image
> 
> You just render text on the bottom of the screen and thats it

im a bit new to c++ so not sure how to render the text

**Comment by twangsriprasoet** [comment]
> im a bit new to c++ so not sure how to render the text

Render the text like watermark does.

**Comment by twangsriprasoet** [comment]
i got it like this.

**Comment by koon666** [comment]
can you give me the code for that?
 https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg

**Comment by NekoRem** [comment]
> can you give me the code for that?
> https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg

You can have this ugly ass code that i made in 1 minute
`void Visuals::showVelocity() noexcept
{
    if (!config->visuals.showVelocity)
        return;

    if (!localPlayer || !localPlayer->isAlive())
        return;

    double velocity = localPlayer->velocity().length2D();
    std::wstring velocityfinal = std::to_wstring(velocity);

    interfaces->surface->setTextFont(Surface::font);
    interfaces->surface->setTextColor(255, 255, 255);
    const auto [width, height] = interfaces->surface->getScreenSize();
    interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
    interfaces->surface->printText(velocityfinal);
}`

**Comment by koon666** [comment]
already help nekorem, I need a code for the speed lines, if anyone knows how to make a code like this, send me pls. EXAMPLE: https://prnt.sc/t8083p

**Comment by NekoRem** [comment]
> already help nekorem, I need a code for the speed lines, if anyone knows how to make a code like this, send me pls. EXAMPLE: https://prnt.sc/t8083p

you make my brain hurt...

**Comment by koon666** [comment]
srry mekorem ;(

**Comment by barbadully** [comment]
> > can you give me the code for that?
> > https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg
> 
> You can have this ugly ass code that i made in 1 minute
> 
> ```
> void Visuals::showVelocity() noexcept
> {
>     if (!config->visuals.showVelocity)
>         return;
> 
>     if (!localPlayer || !localPlayer->isAlive())
>         return;
> 
>     double velocity = localPlayer->velocity().length2D();
>     std::wstring velocityfinal = std::to_wstring(velocity);
> 
>     interfaces->surface->setTextFont(Surface::font);
>     interfaces->surface->setTextColor(255, 255, 255);
>     const auto [width, height] = interfaces->surface->getScreenSize();
>     interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
>     interfaces->surface->printText(velocityfinal);
> }
> ```

could you help me how to use the getscreensize? in example I just want to add some text above my spectator list...

**Comment by NekoRem** [comment]
> > > can you give me the code for that?
> > > https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg
> > 
> > 
> > You can have this ugly ass code that i made in 1 minute
> > ```
> > void Visuals::showVelocity() noexcept
> > {
> >     if (!config->visuals.showVelocity)
> >         return;
> > 
> >     if (!localPlayer || !localPlayer->isAlive())
> >         return;
> > 
> >     double velocity = localPlayer->velocity().length2D();
> >     std::wstring velocityfinal = std::to_wstring(velocity);
> > 
> >     interfaces->surface->setTextFont(Surface::font);
> >     interfaces->surface->setTextColor(255, 255, 255);
> >     const auto [width, height] = interfaces->surface->getScreenSize();
> >     interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
> >     interfaces->surface->printText(velocityfinal);
> > }
> > ```
> 
> could you help me how to use the getscreensize? in example I just want to add some text above my spectator list...

its used in these two lines 
```
const auto [width, height] = interfaces->surface->getScreenSize();
interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
```

**Comment by barbadully** [comment]
thats what I wanted, no copy paste ready shit, just an explanation of where to look

**Comment by xXTOO-HARDXx** [comment]
> > > > can you give me the code for that?
> > > > https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg
> > > 
> > > 
> > > You can have this ugly ass code that i made in 1 minute
> > > ```
> > > void Visuals::showVelocity() noexcept
> > > {
> > >     if (!config->visuals.showVelocity)
> > >         return;
> > > 
> > >     if (!localPlayer || !localPlayer->isAlive())
> > >         return;
> > > 
> > >     double velocity = localPlayer->velocity().length2D();
> > >     std::wstring velocityfinal = std::to_wstring(velocity);
> > > 
> > >     interfaces->surface->setTextFont(Surface::font);
> > >     interfaces->surface->setTextColor(255, 255, 255);
> > >     const auto [width, height] = interfaces->surface->getScreenSize();
> > >     interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
> > >     interfaces->surface->printText(velocityfinal);
> > > }
> > > ```
> > 
> > 
> > could you help me how to use the getscreensize? in example I just want to add some text above my spectator list...
> 
> its used in these two lines the rest is you needing to use your brain for once in your life
> 
> ```
> const auto [width, height] = interfaces->surface->getScreenSize();
> interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
> ```

no need to get toxic. this isnt UC

**Comment by daisuki333** [comment]
> > ```
> > void Visuals::showVelocity() noexcept
> > {
> >     if (!config->visuals.showVelocity)
> >         return;
> > 
> >     if (!localPlayer || !localPlayer->isAlive())
> >         return;
> > 
> >     double velocity = localPlayer->velocity().length2D();
> >     std::wstring velocityfinal = std::to_wstring(velocity);
> > 
> >     interfaces->surface->setTextFont(Surface::font);
> >     interfaces->surface->setTextColor(255, 255, 255);
> >     const auto [width, height] = interfaces->surface->getScreenSize();
> >     interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
> >     interfaces->surface->printText(velocityfinal);
> > }
> > ```

This code doesn't show the velocity for me, idk why it doesn't work, help would be appreciated.

**Comment by koahook** [comment]
> > > > can you give me the code for that?
> > > > https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg
> > > 
> > > 
> > > You can have this ugly ass code that i made in 1 minute
> > > ```
> > > void Visuals::showVelocity() noexcept
> > > {
> > >     if (!config->visuals.showVelocity)
> > >         return;
> > > 
> > >     if (!localPlayer || !localPlayer->isAlive())
> > >         return;
> > > 
> > >     double velocity = localPlayer->velocity().length2D();
> > >     std::wstring velocityfinal = std::to_wstring(velocity);
> > > 
> > >     interfaces->surface->setTextFont(Surface::font);
> > >     interfaces->surface->setTextColor(255, 255, 255);
> > >     const auto [width, height] = interfaces->surface->getScreenSize();
> > >     interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
> > >     interfaces->surface->printText(velocityfinal);
> > > }
> > > ```
> > 
> > 
> > could you help me how to use the getscreensize? in example I just want to add some text above my spectator list...
> 
> its used in these two lines the rest is you needing to use your brain for once in your life
> 
> ```
> const auto [width, height] = interfaces->surface->getScreenSize();
> interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
> ```

Can you help me make this work? I am sort of confused on why it will not show up

**Comment by NekoRem** [comment]
Did you put call the function in paint traverse?

**Comment by synvy** [comment]
> Can you help me make this work? I am sort of confused on why it will not show up

Go learn C++ before trying to make/mess with cheats.  You obviously have no idea what you're doing and if you passed 5th grade the math shouldn't be a problem.

**Comment by braken1** [comment]
hello everytime i turn i jb with my bind key it turns off bunny hop feature how do i fix this?

**Comment by newcommerdontblame** [comment]
> hello everytime i turn i jb with my bind key it turns off bunny hop feature how do i fix this?

static bool bhopWasEnabled = false;

**Comment by braken1** [comment]
> > hello everytime i turn i jb with my bind key it turns off bunny hop feature how do i fix this?
> 
> static bool bhopWasEnabled = false;

i fixed it somehow but now jb dont works as good as it worked before

**Comment by braken1** [comment]
> > > hello everytime i turn i jb with my bind key it turns off bunny hop feature how do i fix this?
> > 
> > 
> > static bool bhopWasEnabled = false;
> 
> i fixed it somehow but now jb dont works as good as it worked before

fixed it nwm

**Comment by SapphireCodenz** [comment]
> > > > hello everytime i turn i jb with my bind key it turns off bunny hop feature how do i fix this?
> > > 
> > > 
> > > static bool bhopWasEnabled = false;
> > 
> > 
> > i fixed it somehow but now jb dont works as good as it worked before
> 
> fixed it nwm

send me the fix on my dc please Toyota AE86#2279

**Comment by braken1** [comment]
how to remove the .0000 from the velocity? also can how to make text outline?

**Comment by NekoRem** [comment]
#1846

**Comment by Jimmy-Baby** [comment]
> how to remove the .0000 from the velocity? also how to make text outline?

You can use format args. Use float precision "%.1f"

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -751,3 +751,101 @@ void Misc::purchaseList(GameEvent* event) noexcept
...
+
+        if (config->misc.bunnyHop) {
+            config->misc.bunnyHop = false;
+            bhopWasEnabled = true;
+        }
+
+        if (unduck) {
+            JumpDone = false;
+            cmd->buttons &= ~UserCmd::IN_DUCK;
+			// cmd->buttons |= UserCmd::IN_JUMP; // If you want to hold JB key only.
```
Comment (playday3008) [inline Osiris/Hacks/Misc.cpp:778]: you can do like this
```cpp
            if (config->misc.jumpbughold)
                cmd->buttons |= UserCmd::IN_JUMP; // If you want to hold JB key only.
            else
                cmd->buttons &= ~UserCmd::IN_DUCK;
```

Comment (playday3008) [reply Osiris/Hacks/Misc.cpp:778]: and in `GUI.cpp` add (after `hotkey(config->misc.jumpbugkey);`)
```cpp
    if (config->misc.jumpbug) {
        ImGui::SameLine();
        ImGui::Checkbox("Hold", &config->misc.jumpbughold);
    }
```

**Comment by newcommerdontblame** [comment]
Hey can i paste edgebug too?

**Comment by Matt0-dev** [comment]
This simply doesnt work for me maybe because there was a update but i have no clue onto why its not working. I even got the old code from UC and tried transferring from the original sdk to osiris and all it does is duck i got it to work like once but that was for only one jump. Does this occur for anyone else and is this why its still open and not in the cheat?

**Comment by Matt0-dev** [comment]
> This simply doesnt work for me maybe because there was a update but i have no clue onto why its not working. I even got the old code from UC and tried transferring from the original sdk to osiris and all it does is duck i got it to work like once but that was for only one jump. Does this occur for anyone else and is this why its still open and not in the cheat?
> 
> `void Misc::jumpbug(UserCmd* cmd) noexcept {
> if (!config->misc.jumpbug || !localPlayer || !localPlayer->isAlive())
> return;
> 
> ```
> static bool bhopWasEnabled = false;
> bool JumpDone;
> 
> auto unduck = true;
> 
> float max_radias = M_PI * 2;
> float step = max_radias / 128;
> float xThick = 23;
> 
> if (GetAsyncKeyState(config->misc.jumpbugkey) && (localPlayer->flags() & 1)) {
> 
>     if (config->misc.bunnyHop) {
>         config->misc.bunnyHop = false;
>         bhopWasEnabled = true;
>     }
> 
>     if (unduck) {
>         JumpDone = false;
>         cmd->buttons |= ~UserCmd::IN_DUCK;
>         unduck = false;
>     }
>     Vector pos = localPlayer->origin();
>     for (float a = 0.f; a < max_radias; a += step) {
>         Vector pt;
>         pt.x = (xThick * cos(a)) + pos.x;
>         pt.y = (xThick * sin(a)) + pos.y;
>         pt.z = pos.z;
> 
>         Vector pt2 = pt;
>         pt2.z -= 6;
> 
>         Trace fag;
> 
>         TraceFilter flt = localPlayer.get();
> 
>         interfaces->engineTrace->traceRay({ pt, pt2 }, 0x1400B, flt, fag);
> 
>         if (fag.fraction != 1.0f && fag.fraction != 0.0f) {
>             JumpDone = true;
>             cmd->buttons |= UserCmd::IN_DUCK;
>             cmd->buttons &= ~UserCmd::IN_JUMP;
>             unduck = true;
>         }
>     }
>     for (float a = 0.f; a < max_radias; a += step) {
>         Vector pt;
>         pt.x = ((xThick - 2.f) * cos(a)) + pos.x;
>         pt.y = ((xThick - 2.f) * sin(a)) + pos.y;
>         pt.z = pos.z;
> 
>         Vector pt2 = pt;
>         pt2.z -= 6;
> 
>         Trace fag;
> 
>         TraceFilter flt = localPlayer.get();
>         interfaces->engineTrace->traceRay({ pt, pt2 }, 0x1400B, flt, fag);
> 
>         if (fag.fraction != 1.f && fag.fraction != 0.f) {
>             JumpDone = true;
>             cmd->buttons |= UserCmd::IN_DUCK;
>             cmd->buttons &= ~UserCmd::IN_JUMP;
>             unduck = true;
>         }
>     }
>     for (float a = 0.f; a < max_radias; a += step) {
>         Vector pt;
>         pt.x = ((xThick - 20.f) * cos(a)) + pos.x;
>         pt.y = ((xThick - 20.f) * sin(a)) + pos.y;
>         pt.z = pos.z;
> 
>         Vector pt2 = pt;
>         pt2.z -= 6;
> 
>         Trace fag;
> 
>         TraceFilter flt = localPlayer.get();
>         interfaces->engineTrace->traceRay({ pt, pt2 }, 0x1400B, flt, fag);
> 
>         if (fag.fraction != 1.f && fag.fraction != 0.f) {
>             JumpDone = true;
>             cmd->buttons |= UserCmd::IN_DUCK;
>             cmd->buttons &= ~UserCmd::IN_JUMP;
>             unduck = true;
>         }
>     }
> }
> else if (bhopWasEnabled) {
>     config->misc.bunnyHop = true;
>     bhopWasEnabled = false;
> }
> ```
> 
> }`

nvm fixed it lol

**Comment by oreothegod** [comment]
> > can you give me the code for that?
> > https://user-images.githubusercontent.com/46624417/85952150-87ebde80-b991-11ea-813c-1b193576e1e8.jpg
> 
> You can have this ugly ass code that i made in 1 minute
> 
> ```
> void Visuals::showVelocity() noexcept
> {
>     if (!config->visuals.showVelocity)
>         return;
> 
>     if (!localPlayer || !localPlayer->isAlive())
>         return;
> 
>     double velocity = localPlayer->velocity().length2D();
>     std::wstring velocityfinal = std::to_wstring(velocity);
> 
>     interfaces->surface->setTextFont(Surface::font);
>     interfaces->surface->setTextColor(255, 255, 255);
>     const auto [width, height] = interfaces->surface->getScreenSize();
>     interfaces->surface->setTextPosition(width / 2 - 16 , height - 200);
>     interfaces->surface->printText(velocityfinal);
> }
> ```

hey I'm a new comer to cheating and I wanted to ask how can I add showVelocity as a member to the class Config::Visuals

**Comment by NekoRem** [comment]
Please close this its outdated. And I don't like getting notifications from it

**Pull request closed without merging**